Zend/zend_inheritance.c

```diff
@@ -570,6 +572,19 @@ static inheritance_status zend_is_class_subtype_of_type(
 	if (have_unresolved) {
 		return INHERITANCE_UNRESOLVED;
 	}
+
+	// replacing static with self in final classes is okay
+	if (!is_intersection && (fe_scope->ce_flags & ZEND_ACC_FINAL) &&
+		(proto_type_full_mask & MAY_BE_STATIC) && instanceof_function(fe_scope, proto_scope)) {
+		if (!fe_ce) fe_ce = lookup_class(fe_scope, fe_class_name);
+
+		if (fe_ce && instanceof_function(fe_ce, fe_scope)) {
```

Comment:
Shouldn't this use `unlinked_instanceof` ?
Actually, can't this entire if condition be `fe_ce == fe_scope` ?